spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/discovery/NacosWatch.java

```diff
@@ -153,7 +153,7 @@ private void resetIfNeeded(Instance instance) {
 
 	private Optional<Instance> selectCurrentInstance(List<Instance> instances) {
 		return instances.stream()
-				.filter(instance -> properties.getIp().equals(instance.getIp())
+				.filter(instance -> properties.getHost().equals(instance.getIp())
```

Comment:
IP is a more specific field, and host is not particularly clear, so we should try to use IP instead of host, but you, on the contrary, changed all the places where IP addresses were originally used to host, which is unreasonable. Adding host is just to satisfy some small number of users. I think the core logic should be that the user configures the IP and the host to use the IP. If the IP is not configured, only the Host is configured, and the host is converted to an IP and set in the IP field.

Reply:
the pr's target is we can use hostname to register service. if use hostname,  the ip field's value is hostname.   it's ok?

Reply:
@steverao  as the #1403 describe： This meant that I had to configure the hostname or domain name for the server to bind to the private cloud IP for communication, but I checked the documentation and found that only Erauke provided this parameter. Although the InetUtils SpringCloud class USES InetUtilsProperties classes to do some configuration Settings, but still can't achieve my goal, in the meantime I use the ignoredInterfaces parameters (value is' [\ s \ s] * '), but findFirstNonLoopbackAddress () calls the InetAddress. Eventually getLocalHost ().
这意味着我必须为服务器配置hostname 或者域名的方式来绑定私有云IP 来进行通信，但是我查阅了文档发现，只有Erauke 提供这种参数。
虽然SpringCloud中的InetUtils类使用了InetUtilsProperties类来进行一些配置化的设置，但依旧不能达到我的目的，在此期间我使用了ignoredInterfaces 参数（值是'[\s\S]*'），但是findFirstNonLoopbackAddress() 最终还是调用了InetAddress.getLocalHost()。

Reply:
the host can be hostname, domain or ip,   and register the service registry.

Reply:
> @steverao as the #1403 describe： This meant that I had to configure the hostname or domain name for the server to bind to the private cloud IP for communication, but I checked the documentation and found that only Erauke provided this parameter. Although the InetUtils SpringCloud class USES InetUtilsProperties classes to do some configuration Settings, but still can't achieve my goal, in the meantime I use the ignoredInterfaces parameters (value is' [\ s \ s] * '), but findFirstNonLoopbackAddress () calls the InetAddress. Eventually getLocalHost (). 这意味着我必须为服务器配置hostname 或者域名的方式来绑定私有云IP 来进行通信，但是我查阅了文档发现，只有Erauke 提供这种参数。 虽然SpringCloud中的InetUtils类使用了InetUtilsProperties类来进行一些配置化的设置，但依旧不能达到我的目的，在此期间我使用了ignoredInterfaces 参数（值是'[\s\S]*'），但是findFirstNonLoopbackAddress() 最终还是调用了InetAddress.getLocalHost()。

From that issue, I think this is the purpose: 
So back to the NacosDiscovery，NacosDiscovery provides spring.cloud.nacos.discovery.ip parameter (this parameter can fill in the form of a hostname), but I think this parameter should be dynamically acquired rather than specified, because our environment might have dev, test, pro... Although shell scripts in various environments can be specified, this is still tedious work.
再说回NacosDiscovery，NacosDiscovery 提供了spring.cloud.nacos.discovery.ip 参数（这个参数可以填写为hostname 的形式），但是我觉得这个参数应该是动态获取而不是指定，因为我们的环境可能会有dev、test、pro...，尽管在各个环境的shell 脚本能够指定，但这工作依然很繁琐。

Reply:
> the pr's target is we can use hostname to register service. if use hostname, the ip field's value is hostname. it's ok?

I think ip field's value is a IP address extracted from hostname, if you feel confused about the demand you can leave message in that issue and communicate with author.

Reply:
> > @steverao as the #1403 describe： This meant that I had to configure the hostname or domain name for the server to bind to the private cloud IP for communication, but I checked the documentation and found that only Erauke provided this parameter. Although the InetUtils SpringCloud class USES InetUtilsProperties classes to do some configuration Settings, but still can't achieve my goal, in the meantime I use the ignoredInterfaces parameters (value is' [\ s \ s] * '), but findFirstNonLoopbackAddress () calls the InetAddress. Eventually getLocalHost (). 这意味着我必须为服务器配置hostname 或者域名的方式来绑定私有云IP 来进行通信，但是我查阅了文档发现，只有Erauke 提供这种参数。 虽然SpringCloud中的InetUtils类使用了InetUtilsProperties类来进行一些配置化的设置，但依旧不能达到我的目的，在此期间我使用了ignoredInterfaces 参数（值是'[\s\S]*'），但是findFirstNonLoopbackAddress() 最终还是调用了InetAddress.getLocalHost()。
> 
> From that issue, I think this is the purpose: So back to the NacosDiscovery，NacosDiscovery provides spring.cloud.nacos.discovery.ip parameter (this parameter can fill in the form of a hostname), but I think this parameter should be dynamically acquired rather than specified, because our environment might have dev, test, pro... Although shell scripts in various environments can be specified, this is still tedious work. 再说回NacosDiscovery，NacosDiscovery 提供了spring.cloud.nacos.discovery.ip 参数（这个参数可以填写为hostname 的形式），但是我觉得这个参数应该是动态获取而不是指定，因为我们的环境可能会有dev、test、pro...，尽管在各个环境的shell 脚本能够指定，但这工作依然很繁琐。

in this case, the hostname must specific, using  'spring.cloud.nacos.discovery.ip'. if the  'spring.cloud.nacos.discovery.ip' is specific, the ip's acture value is hostname, and there is no more checkes or convert.   i want the hostname can be auto detected.